jlm/rvsdg/theta.hpp

```diff
@@ -7,6 +7,8 @@
 #ifndef JLM_RVSDG_THETA_HPP
 #define JLM_RVSDG_THETA_HPP
 
+#include <optional>
+
```

Comment:
I normally put std library include files below the jlm include files. Is there a particular reason why you put them above, or just coincidence?

Reply:
normally I do it in this order: C headers, C++ language headers, external C++ library headers, then project headers; rationale is that this is also the order of precedence / dependence

but sorted it as requested